File: mobile/app/salary-calculator.js
// Take-home pay calculator. Type your monthly gross and see the SSS,
// PhilHealth, Pag-IBIG, and income tax taken out, and what lands in your
// pocket. All math is in lib/phtax.js (pure and tested). This is an estimate
// from published rates, not your official payslip, and the screen says so.

import { useMemo, useState } from 'react';
import { Pressable, ScrollView, StyleSheet, Text, TextInput, View } from 'react-native';
import { SafeAreaView } from 'react-native-safe-area-context';
import { useRouter } from 'expo-router';
import { Ionicons } from '@expo/vector-icons';
import { spacing, radius, fontSize, fontWeight } from '../theme';
import { useTheme } from '../context/Theme';
import { formatMoney } from '../lib/format';
import { takeHomePay, RATES_YEAR } from '../lib/phtax';

export default function SalaryCalculator() {
  const { colors } = useTheme();
  const styles = useMemo(() => makeStyles(colors), [colors]);
  const router = useRouter();

  const [basic, setBasic] = useState('');
  const [taxAllow, setTaxAllow] = useState('');
  const [nonTaxAllow, setNonTaxAllow] = useState('');
  const [period, setPeriod] = useState('month');
  const parse = (s) => Number(String(s).replace(/[, ]/g, '')) || 0;
  const basicNum = parse(basic);
  const taxAllowNum = parse(taxAllow);
  const nonTaxAllowNum = parse(nonTaxAllow);
  const r = useMemo(
    () => takeHomePay(basicNum, { taxableAllowance: taxAllowNum, nonTaxableAllowance: nonTaxAllowNum }),
    [basicNum, taxAllowNum, nonTaxAllowNum]
  );
  const m = (n) => formatMoney(Math.round(n));

  // The engine works in monthly pesos. The user can view the result per
  // semi-monthly cutoff (half), monthly, or annualized (times twelve). The
  // basic pay input stays monthly; only the breakdown figures are scaled.
  const PERIODS = [
    { id: 'cutoff', label: 'Per cutoff', factor: 0.5, word: 'per cutoff' },
    { id: 'month', label: 'Monthly', factor: 1, word: 'a month' },
    { id: 'year', label: 'Per year', factor: 12, word: 'a year' },
  ];
  const active = PERIODS.find((p) => p.id === period) || PERIODS[1];
  const ms = (n) => m(n * active.factor);
  const netLabel = period === 'cutoff' ? 'Take-home per cutoff' : period === 'year' ? 'Take-home per year' : 'Take-home pay';

  const rows = [
    { label: 'SSS', value: r.sss },
    { label: 'PhilHealth', value: r.philhealth },
    { label: 'Pag-IBIG', value: r.pagibig },
    { label: 'Income tax', value: r.monthlyTax },
  ];

  return (
    <SafeAreaView style={styles.screen} edges={['top']}>
      <View style={styles.headerBar}>
        <Pressable onPress={() => router.back()} hitSlop={10} style={styles.back}>
          <Ionicons name="chevron-back" size={24} color={colors.text} />
        </Pressable>
        <Text style={styles.headerTitle}>Take-home pay</Text>
        <View style={{ width: 24 }} />
      </View>

      <ScrollView contentContainerStyle={styles.content} keyboardShouldPersistTaps="handled">
        <Text style={styles.fieldLabel}>Monthly basic pay</Text>
        <View style={styles.inputWrap}>
          <Text style={styles.peso}>₱</Text>
          <TextInput
            style={styles.input}
            value={basic}
            onChangeText={setBasic}
            keyboardType="numeric"
            placeholder="e.g. 25,000"
            placeholderTextColor={colors.faint}
            autoFocus
          />
        </View>

        <View style={styles.allowRow}>
          <View style={styles.allowCol}>
            <Text style={styles.fieldLabel}>Taxable allowance</Text>
            <View style={styles.smallInputWrap}>
              <Text style={styles.pesoSm}>₱</Text>
              <TextInput style={styles.smallInput} value={taxAllow} onChangeText={setTaxAllow} keyboardType="numeric" placeholder="0" placeholderTextColor={colors.faint} />
            </View>
          </View>
          <View style={styles.allowCol}>
            <Text style={styles.fieldLabel}>Non-taxable allowance</Text>
            <View style={styles.smallInputWrap}>
              <Text style={styles.pesoSm}>₱</Text>
              <TextInput style={styles.smallInput} value={nonTaxAllow} onChangeText={setNonTaxAllow} keyboardType="numeric" placeholder="0" placeholderTextColor={colors.faint} />
            </View>
          </View>
        </View>
        <Text style={styles.allowHint}>
          Non-taxable covers de minimis benefits and allowances within BIR limits. They are added to your pay but not taxed. Contributions are figured on your basic pay.
        </Text>

        {basicNum > 0 && r.net >= 0 ? (
          <>
            <Text style={styles.fieldLabel}>Show results</Text>
            <View style={styles.segment}>
              {PERIODS.map((p) => (
                <Pressable key={p.id} style={[styles.segBtn, period === p.id && styles.segBtnOn]} onPress={() => setPeriod(p.id)}>
                  <Text style={[styles.segText, period === p.id && styles.segTextOn]}>{p.label}</Text>
                </Pressable>
              ))}
            </View>

            <View style={styles.card}>
              <View style={styles.grossRow}>
                <Text style={styles.grossLabel}>Basic pay</Text>
                <Text style={styles.grossValue}>{ms(r.basic)}</Text>
              </View>
              {r.taxableAllowance > 0 ? (
                <View style={styles.row}>
                  <Text style={styles.rowLabel}>Taxable allowance</Text>
                  <Text style={styles.addValue}>+ {ms(r.taxableAllowance)}</Text>
                </View>
              ) : null}
              {r.nonTaxableAllowance > 0 ? (
                <View style={styles.row}>
                  <Text style={styles.rowLabel}>Non-taxable allowance</Text>
                  <Text style={styles.addValue}>+ {ms(r.nonTaxableAllowance)}</Text>
                </View>
              ) : null}
              {r.taxableAllowance > 0 || r.nonTaxableAllowance > 0 ? (
                <View style={[styles.row, styles.rowTopBorder]}>
                  <Text style={styles.grossLabel}>Gross pay</Text>
                  <Text style={styles.grossValue}>{ms(r.gross)}</Text>
                </View>
              ) : null}
              {rows.map((row, i) => (
                <View key={row.label} style={[styles.row, i === 0 && styles.rowTopBorder]}>
                  <Text style={styles.rowLabel}>{row.label}</Text>
                  <Text style={styles.rowValue}>- {ms(row.value)}</Text>
                </View>
              ))}
              <View style={styles.netRow}>
                <Text style={styles.netLabel}>{netLabel}</Text>
                <Text style={styles.netValue}>{ms(r.net)}</Text>
              </View>
              {period === 'cutoff' ? (
                <Text style={styles.perYear}>An estimate for each of the two cutoffs (15th and end of month). Real cutoff withholding can split a little differently.</Text>
              ) : (
                <Text style={styles.perYear}>About {m(r.net * 12)} a year, before any 13th month.</Text>
              )}
            </View>

            <View style={styles.deductCard}>
              <Text style={styles.deductKicker}>WHAT COMES OUT, AND WHY</Text>
              <Text style={styles.deductLine}>
                Contributions total {ms(r.contributions)} {active.word}. They come out before tax, so your taxable pay is {ms(r.monthlyTaxable)}.
              </Text>
              <Text style={styles.deductLine}>
                Income tax uses the graduated BIR table on your yearly taxable pay, spread across the year.
              </Text>
            </View>
          </>
        ) : basicNum > 0 ? (
          <Text style={styles.hint}>
            That looks too low for a monthly salary. The minimum SSS and PhilHealth contributions alone come to about {m(r.contributions)}, so a basic pay under that would leave nothing to take home. Enter your full monthly basic pay.
          </Text>
        ) : (
          <Text style={styles.hint}>Enter your monthly basic pay to see the breakdown. Allowances add on top of it.</Text>
        )}

        <Text style={styles.disclaimer}>
          Estimate based on {RATES_YEAR} SSS, PhilHealth, Pag-IBIG, and BIR rates. Contributions are figured on your basic pay, non-taxable allowances are not taxed, and low salaries still pay the minimum contributions. Your real payslip can differ with de minimis limits and your employer's rounding. Not a substitute for your official payslip or a BIR filing.
        </Text>
      </ScrollView>
    </SafeAreaView>
  );
}

function makeStyles(colors) {
  return StyleSheet.create({
    screen: { flex: 1, backgroundColor: colors.background },
    headerBar: { flexDirection: 'row', alignItems: 'center', justifyContent: 'space-between', paddingHorizontal: spacing.lg, paddingTop: spacing.md, paddingBottom: spacing.sm },
    back: { marginLeft: -4 },
    headerTitle: { color: colors.text, fontSize: fontSize.subtitle, fontWeight: fontWeight.bold },
    content: { padding: spacing.lg, paddingBottom: spacing.xxl },

    fieldLabel: { color: colors.muted, fontSize: fontSize.caption, marginBottom: spacing.sm, letterSpacing: 0.3 },
    inputWrap: { flexDirection: 'row', alignItems: 'center', backgroundColor: colors.card, borderColor: colors.border, borderWidth: 1, borderRadius: radius.md, paddingHorizontal: spacing.lg, marginBottom: spacing.lg },
    peso: { color: colors.muted, fontSize: fontSize.title, fontWeight: fontWeight.bold, marginRight: spacing.sm },
    input: { flex: 1, paddingVertical: spacing.md, color: colors.text, fontSize: fontSize.title, fontWeight: fontWeight.bold },

    allowRow: { flexDirection: 'row', gap: spacing.md, marginBottom: spacing.sm },
    allowCol: { flex: 1 },
    smallInputWrap: { flexDirection: 'row', alignItems: 'center', backgroundColor: colors.card, borderColor: colors.border, borderWidth: 1, borderRadius: radius.md, paddingHorizontal: spacing.md },
    pesoSm: { color: colors.muted, fontSize: fontSize.body, marginRight: spacing.xs },
    smallInput: { flex: 1, paddingVertical: spacing.md, color: colors.text, fontSize: fontSize.body, fontWeight: fontWeight.medium },
    allowHint: { color: colors.faint, fontSize: fontSize.caption, lineHeight: 16, marginBottom: spacing.lg },
    addValue: { color: colors.textSecondary, fontSize: fontSize.small, fontWeight: fontWeight.medium },

    segment: { flexDirection: 'row', backgroundColor: colors.card, borderColor: colors.border, borderWidth: 1, borderRadius: radius.md, padding: 3, marginBottom: spacing.lg },
    segBtn: { flex: 1, paddingVertical: spacing.sm, alignItems: 'center', borderRadius: radius.sm },
    segBtnOn: { backgroundColor: colors.primary },
    segText: { color: colors.muted, fontSize: fontSize.small, fontWeight: fontWeight.medium },
    segTextOn: { color: colors.background, fontWeight: fontWeight.bold },

    card: { backgroundColor: colors.card, borderColor: colors.border, borderWidth: 1, borderRadius: radius.lg, padding: spacing.lg, marginBottom: spacing.lg },
    grossRow: { flexDirection: 'row', justifyContent: 'space-between', alignItems: 'center', paddingBottom: spacing.md },
    grossLabel: { color: colors.text, fontSize: fontSize.body, fontWeight: fontWeight.medium },
    grossValue: { color: colors.text, fontSize: fontSize.body, fontWeight: fontWeight.bold },
    row: { flexDirection: 'row', justifyContent: 'space-between', alignItems: 'center', paddingVertical: spacing.sm },
    rowTopBorder: { borderTopColor: colors.border, borderTopWidth: StyleSheet.hairlineWidth, paddingTop: spacing.md },
    rowLabel: { color: colors.muted, fontSize: fontSize.small },
    rowValue: { color: colors.textSecondary, fontSize: fontSize.small, fontWeight: fontWeight.medium },
    netRow: { flexDirection: 'row', justifyContent: 'space-between', alignItems: 'center', borderTopColor: colors.border, borderTopWidth: 1, marginTop: spacing.md, paddingTop: spacing.md },
    netLabel: { color: colors.text, fontSize: fontSize.subtitle, fontWeight: fontWeight.bold },
    netValue: { color: colors.primary, fontSize: fontSize.subtitle, fontWeight: fontWeight.heavy },
    perYear: { color: colors.faint, fontSize: fontSize.caption, marginTop: spacing.sm },

    deductCard: { backgroundColor: colors.card, borderColor: colors.border, borderWidth: 1, borderRadius: radius.lg, padding: spacing.lg, marginBottom: spacing.lg },
    deductKicker: { color: colors.muted, fontSize: fontSize.caption, fontWeight: fontWeight.bold, letterSpacing: 1.2, marginBottom: spacing.sm },
    deductLine: { color: colors.textSecondary, fontSize: fontSize.small, lineHeight: 19, marginTop: spacing.xs },

    hint: { color: colors.muted, fontSize: fontSize.body, textAlign: 'center', marginTop: spacing.xl, marginBottom: spacing.xl },
    disclaimer: { color: colors.faint, fontSize: fontSize.caption, lineHeight: 17, marginTop: spacing.sm },
  });
}

File: flutter/lib/screens/salary_calculator.dart
// Take-home pay: gross to net for a PH employee, adapted from the RN
// screen on the golden-ported phtax engine (SSS, PhilHealth, Pag-IBIG, and
// the graduated BIR table). The engine works in monthly pesos; the user can
// view the result per cutoff, monthly, or per year. Every explanation the
// RN screen gives about WHY the money comes out rides along.

import 'package:flutter/material.dart';

import '../money/phtax.dart';
import '../theme.dart';
import 'overview.dart' show formatMoney;

class SalaryCalculatorScreen extends StatefulWidget {
  const SalaryCalculatorScreen({super.key});

  @override
  State<SalaryCalculatorScreen> createState() =>
      _SalaryCalculatorScreenState();
}

class _SalaryCalculatorScreenState extends State<SalaryCalculatorScreen> {
  final basic = TextEditingController();
  final taxAllow = TextEditingController();
  final nonTaxAllow = TextEditingController();
  String period = 'month';

  @override
  void dispose() {
    basic.dispose();
    taxAllow.dispose();
    nonTaxAllow.dispose();
    super.dispose();
  }

  double _parse(String s) {
    final cleaned = s.replaceAll(RegExp(r'[, ]'), '');
    if (cleaned.isEmpty) return 0;
    final n = double.tryParse(cleaned);
    return (n == null || !n.isFinite) ? 0 : n;
  }

  String _m(num n) => formatMoney((n + 0.5).floorToDouble());

  Widget _label(String text) => Padding(
        padding: const EdgeInsets.only(top: 14, bottom: 6),
        child: Text(text,
            style: TextStyle(
                color: Barako.muted,
                fontSize: 12,
                fontWeight: FontWeight.w700)),
      );

  Widget _line(String label, String value,
      {bool strong = false, Color? valueColor}) {
    return Padding(
      padding: const EdgeInsets.symmetric(vertical: 3),
      child: Row(
        children: [
          Expanded(
            child: Text(label,
                style: TextStyle(
                    color: strong ? Barako.text : Barako.textSecondary,
                    fontSize: 13,
                    fontWeight:
                        strong ? FontWeight.w700 : FontWeight.w400)),
          ),
          Text(value,
              style: TextStyle(
                  color: valueColor ??
                      (strong ? Barako.text : Barako.textSecondary),
                  fontSize: strong ? 15 : 13,
                  fontWeight: strong ? FontWeight.w800 : FontWeight.w600,
                  fontFeatures: const [FontFeature.tabularFigures()])),
        ],
      ),
    );
  }

  @override
  Widget build(BuildContext context) {
    final basicNum = _parse(basic.text);
    final taxAllowNum = _parse(taxAllow.text);
    final nonTaxAllowNum = _parse(nonTaxAllow.text);
    final r = takeHomePay(basicNum,
        taxableAllowance: taxAllowNum, nonTaxableAllowance: nonTaxAllowNum);

    // The engine works in monthly pesos; only the displayed figures scale.
    final factor = period == 'cutoff' ? 0.5 : period == 'year' ? 12.0 : 1.0;
    final word = period == 'cutoff'
        ? 'per cutoff'
        : period == 'year'
            ? 'a year'
            : 'a month';
    String ms(num n) => _m(n * factor);
    final netLabel = period == 'cutoff'
        ? 'Take-home per cutoff'
        : period == 'year'
            ? 'Take-home per year'
            : 'Take-home pay';

    final gross = r['gross'] as double;
    final monthlyTax = r['monthlyTax'] as double;
    // Effective rate is tax as a share of gross; marginal is the bracket the
    // next taxable peso falls in. Both help the user read a raise correctly.
    final effRate = gross > 0
        ? ((monthlyTax / gross * 1000) + 0.5).floorToDouble() / 10
        : 0.0;
    final effRateText = effRate == 0 && monthlyTax > 0
        ? 'less than 0.1%'
        : '${effRate % 1 == 0 ? effRate.toInt() : effRate}%';
    final marginal =
        ((marginalRate((r['monthlyTaxable'] as double) * 12) * 100) + 0.5)
            .floorToDouble()
            .toInt();

    final showResults = basicNum > 0 && (r['net'] as double) >= 0;
    final tooLow = basicNum > 0 && !showResults;

    return Scaffold(
      appBar: AppBar(
        backgroundColor: Barako.background,
        foregroundColor: Barako.text,
        title: Text('Take-home pay',
            style:
                TextStyle(color: Barako.text, fontWeight: FontWeight.w800)),
      ),
      body: SafeArea(
        child: ListView(
          padding: const EdgeInsets.fromLTRB(20, 8, 20, 32),
          children: [
            _label('MONTHLY BASIC PAY'),
            TextField(
              controller: basic,
              onChanged: (_) => setState(() {}),
              keyboardType:
                  const TextInputType.numberWithOptions(decimal: true),
              autofocus: true,
              style: TextStyle(color: Barako.text),
              decoration: InputDecoration(
                  prefixText: '₱ ', hintText: 'e.g. 25,000'),
            ),
            Row(
              crossAxisAlignment: CrossAxisAlignment.start,
              children: [
                Expanded(
                  child: Column(
                    crossAxisAlignment: CrossAxisAlignment.start,
                    children: [
                      _label('TAXABLE ALLOWANCE'),
                      TextField(
                        controller: taxAllow,
                        onChanged: (_) => setState(() {}),
                        keyboardType: const TextInputType.numberWithOptions(
                            decimal: true),
                        style: TextStyle(color: Barako.text),
                        decoration: InputDecoration(
                            prefixText: '₱ ', hintText: '0'),
                      ),
                    ],
                  ),
                ),
                const SizedBox(width: 12),
                Expanded(
                  child: Column(
                    crossAxisAlignment: CrossAxisAlignment.start,
                    children: [
                      _label('NON-TAXABLE ALLOWANCE'),
                      TextField(
                        controller: nonTaxAllow,
                        onChanged: (_) => setState(() {}),
                        keyboardType: const TextInputType.numberWithOptions(
                            decimal: true),
                        style: TextStyle(color: Barako.text),
                        decoration: InputDecoration(
                            prefixText: '₱ ', hintText: '0'),
                      ),
                    ],
                  ),
                ),
              ],
            ),
            const SizedBox(height: 6),
            Text(
                'Non-taxable covers de minimis benefits and allowances within BIR limits. They are added to your pay but not taxed. Contributions are figured on your basic pay.',
                style: TextStyle(
                    color: Barako.muted, fontSize: 12, height: 1.4)),
            // The RN screen warns above 12,000, but the realistic monthly
            // total of the common de minimis items is 4,000 to 5,000, so
            // the compensation specialist lowered the trigger here.
            if (nonTaxAllowNum > 5000)
              Padding(
                padding: const EdgeInsets.only(top: 8),
                child: Text(
                    'Heads up: each de minimis benefit has its own BIR ceiling. Amounts above those ceilings are not taxed right away, they first join the ₱90,000 a year cap together with your 13th month and other bonuses, and only the part above ₱90,000 gets taxed. A large non-taxable amount here can still overstate your take-home.',
                    style: TextStyle(
                        color: Barako.warning, fontSize: 12, height: 1.4)),
              ),
            if (showResults) ...[
              _label('SHOW RESULTS'),
              Wrap(
                spacing: 8,
                children: [
                  for (final p in const [
                    ('cutoff', 'Per cutoff'),
                    ('month', 'Monthly'),
                    ('year', 'Per year'),
                  ])
                    ChoiceChip(
                      label: Text(p.$2),
                      selected: period == p.$1,
                      onSelected: (_) => setState(() => period = p.$1),
                      selectedColor: Barako.primary,
                      backgroundColor: Barako.background,
                      labelStyle: TextStyle(
                          color: period == p.$1
                              ? Barako.onPrimary
                              : Barako.textSecondary,
                          fontSize: 12,
                          fontWeight: FontWeight.w600),
                    ),
                ],
              ),
              const SizedBox(height: 12),
              Card(
                child: Padding(
                  padding: const EdgeInsets.all(16),
                  child: Column(
                    crossAxisAlignment: CrossAxisAlignment.start,
                    children: [
                      _line('Basic pay', ms(r['basic'] as double),
                          strong: true),
                      if ((r['taxableAllowance'] as double) > 0)
                        _line('Taxable allowance',
                            '+ ${ms(r['taxableAllowance'] as double)}'),
                      if ((r['nonTaxableAllowance'] as double) > 0)
                        _line('Non-taxable allowance',
                            '+ ${ms(r['nonTaxableAllowance'] as double)}'),
                      if ((r['taxableAllowance'] as double) > 0 ||
                          (r['nonTaxableAllowance'] as double) > 0) ...[
                        Divider(color: Barako.border, height: 14),
                        _line('Gross pay', ms(gross), strong: true),
                      ],
                      Divider(color: Barako.border, height: 14),
                      _line('SSS', '- ${ms(r['sss'] as double)}'),
                      _line('PhilHealth',
                          '- ${ms(r['philhealth'] as double)}'),
                      _line('Pag-IBIG', '- ${ms(r['pagibig'] as double)}'),
                      _line('Income tax', '- ${ms(monthlyTax)}'),
                      Divider(color: Barako.border, height: 14),
                      Row(
                        children: [
                          Expanded(
                            child: Text(netLabel,
                                style: TextStyle(
                                    color: Barako.text,
                                    fontSize: 14,
                                    fontWeight: FontWeight.w700)),
                          ),
                          Text(ms(r['net'] as double),
                              style: TextStyle(
                                  color: Barako.primary,
                                  fontSize: 24,
                                  fontFamily: Barako.displayFont,
                                  fontWeight: FontWeight.w700)),
                        ],
                      ),
                      const SizedBox(height: 4),
                      Text(
                          period == 'cutoff'
                              ? 'An estimate for each of the two cutoffs (15th and end of month), splitting everything in half. Many employers deduct SSS, PhilHealth, and Pag-IBIG on just one cutoff, so one payout can be noticeably bigger and the other smaller. The two cutoffs together still match the monthly figure.'
                              : 'About ${_m((r['net'] as double) * 12)} a year, before any 13th month.',
                          style: TextStyle(
                              color: Barako.muted,
                              fontSize: 11,
                              height: 1.4)),
                    ],
                  ),
                ),
              ),
              const SizedBox(height: 12),
              Card(
                child: Padding(
                  padding: const EdgeInsets.all(16),
                  child: Column(
                    crossAxisAlignment: CrossAxisAlignment.start,
                    children: [
                      Text('WHAT COMES OUT, AND WHY',
                          style: TextStyle(
                              color: Barako.muted,
                              fontSize: 10,
                              fontWeight: FontWeight.w700,
                              letterSpacing: 2)),
                      const SizedBox(height: 6),
                      Text(
                          'Contributions total ${ms(r['contributions'] as double)} $word. They come out before tax, so your taxable pay is ${ms(r['monthlyTaxable'] as double)}.',
                          style: TextStyle(
                              color: Barako.textSecondary,
                              fontSize: 12,
                              height: 1.4)),
                      const SizedBox(height: 4),
                      Text(
                          'Income tax uses the graduated BIR table on your yearly taxable pay, spread across the year.',
                          style: TextStyle(
                              color: Barako.textSecondary,
                              fontSize: 12,
                              height: 1.4)),
                    ],
                  ),
                ),
              ),
              const SizedBox(height: 12),
              Card(
                child: Padding(
                  padding: const EdgeInsets.all(16),
                  child: Column(
                    crossAxisAlignment: CrossAxisAlignment.start,
                    children: [
                      Text('YOUR TAX RATE',
                          style: TextStyle(
                              color: Barako.muted,
                              fontSize: 10,
                              fontWeight: FontWeight.w700,
                              letterSpacing: 2)),
                      const SizedBox(height: 6),
                      Text(
                          'Your income tax is about $effRateText of your gross pay.',
                          style: TextStyle(
                              color: Barako.textSecondary,
                              fontSize: 12,
                              height: 1.4)),
                      const SizedBox(height: 4),
                      Text(
                          marginal == 0
                              ? 'Your taxable pay is within the tax-free ₱250,000 a year, so no income tax is due.'
                              : 'You are in the $marginal% tax bracket, so each extra ₱100 of taxable pay is taxed about ₱$marginal. A raise is only taxed at the margin, never your whole pay. Contributions can also rise a bit with a raise until their ceilings.',
                          style: TextStyle(
                              color: Barako.textSecondary,
                              fontSize: 12,
                              height: 1.4)),
                      const SizedBox(height: 4),
                      Text(
                          'If you earn the minimum wage, your basic, overtime, holiday, and night pay are income tax free by law, so your real tax is likely zero.',
                          style: TextStyle(
                              color: Barako.textSecondary,
                              fontSize: 12,
                              height: 1.4)),
                    ],
                  ),
                ),
              ),
            ] else if (tooLow)
              Padding(
                padding: const EdgeInsets.only(top: 12),
                child: Text(
                    'That looks too low for a monthly salary. The minimum SSS, PhilHealth, and Pag-IBIG contributions come to about ${_m(r['contributions'] as double)}, so a basic pay under that would leave nothing to take home. Enter your full monthly basic pay.',
                    style: TextStyle(
                        color: Barako.muted, fontSize: 13, height: 1.4)),
              )
            else
              Padding(
                padding: const EdgeInsets.only(top: 12),
                child: Text(
                    'Enter your monthly basic pay to see the breakdown. Allowances add on top of it.',
                    style: TextStyle(
                        color: Barako.muted, fontSize: 13, height: 1.4)),
              ),
            const SizedBox(height: 12),
            Text(
                'Estimate based on $ratesYear SSS, PhilHealth, Pag-IBIG, and BIR rates. Contributions are figured on your basic pay, non-taxable allowances are not taxed, and low salaries still pay the minimum contributions. It assumes a full month worked with no absences, and no loan payments (SSS or Pag-IBIG salary loans), HMO share, or other company deductions. Your real payslip can differ with de minimis limits and your employer\'s rounding. Not a substitute for your official payslip or a BIR filing.',
                style:
                    TextStyle(color: Barako.faint, fontSize: 11, height: 1.4)),
          ],
        ),
      ),
    );
  }
}
